add userinfo support for OpenID

harshavardhana opened the pull request:
## Description
add userinfo support for OpenID

## Motivation and Context
Some identity providers like GitLab don not provide information
about group membership as part of the identity token. They
only expose it via /oauth/userinfo endpoint, as described
in the OpenID Connect 1.0 sepcification.

fixes #12367

## How to test this PR?
As per the original issue #12367 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Documentation updated
- [ ] Unit tests added/updated

bachp commented on cmd/sts-handlers.go:
```diff
@@ -302,8 +303,9 @@ func (sts *stsAPIHandlers) AssumeRoleWithSSO(w http.ResponseWriter, r *http.Requ
 	if token == "" {
 		token = r.Form.Get(stsWebIdentityToken)
 	}
+	accessToken := r.Form.Get(stsWebIdentityAccessToken)
```
I tested this against our gitlab instance, but the `accessToken` is always empty. Looking at `r.Form` it only contains the `WebIdentityToken` not the `WebIdentityAccessToken`.

harshavardhana replied:
@bachp you need to provide just like `id_token` for JWT - accessToken needs to be provided if UserInfo() support is needed.

bachp replied:
@harshavardhana By provided you mean it should be returned by the OpenID connect provider? Or is there something I need to configure on minio side?

bachp replied:
@harshavardhana Is this something that needs to be implemented on minio/console side?

harshavardhana replied:
MinIO console won't support this just yet, it is only implemented as an API support right now.

Userinfo endpoint cannot use the same id_token for authorization, it needs to be access_token unfortunately.

This is why this API is an extension and is only an opt in. If UserInfo is requested apps have to change

bachp replied:
Ok thanks for clarifying.

bachp reviewed (changes requested):
First test against our gitlab instance didn't work.

harshavardhana committed:
add userinfo support for OpenID

Some identity providers like GitLab do not provide
information about group membership as part of the
identity token claims. They only expose it via OIDC compatible
'/oauth/userinfo' endpoint, as described in the OpenID
Connect 1.0 sepcification.

But this of course requires application to make sure to add
additional accessToken, since idToken cannot be re-used to
perform the same 'userinfo' call. This is why this is specialized
requirement. Gitlab seems to be the only OpenID vendor that requires
this support for the time being.

fixes #12367
